openpifpaf/network/losses/composite.py

```diff
@@ -104,6 +119,8 @@ def _localization_loss(self, x_regs, t_regs):
             )
             if self.prescale != 1.0:
                 loss = loss * self.prescale
+            if self.weights is not None:
+                loss = loss * torch.masked_select(weights_stacked[i, :, :, :, :], reg_masks)
```

Comment:
So I meant inside this for loop, I was unsure if it is okay for the autograd to use the same weight for different vector fields.

Reply:
There are actually two reasons why this is ok to remove:
1) the weights don't have gradients themselves. For Autograd, this is just constant.
2) Module tree != computation graph. One of the magic things that goes on behind the scenes. Think about LSTM cells or anything like this where the same linear layer is re-used over and over with different inputs before `backwards()` is called. PyTorch "unrolls" these operations in the computation graph. This is one of the reasons why training can take so much more GPU memory than prediction.

Reply:
Nice, thank you for that thorough explanation! I will change it now.